pkg/node/ledger/address.go

```diff
@@ -55,17 +55,26 @@ func Addresses(client *node.Client, addr []string) ([]Address, error) {
 	return content, nil
 }
 
-func KeysFiltered(client *node.Client, scAddress string, keyPrefix string) ([]string, error) {
+// KeyFiltered returns an array of Key in byte array filtered with a prefix.
+// If isIncluded is true, will return all the keys with the given prefix,
+// If isIncluded is false, will return all the keys without the given prefix.
+func KeysFiltered(client *node.Client, scAddress string, keyPrefix string, isIncluded bool) ([][]byte, error) {
 	results, err := Addresses(client, []string{scAddress})
 	if err != nil {
 		return nil, fmt.Errorf("calling get_addresses with '%+v': %w", scAddress, err)
 	}
 
-	var filteredKeys []string
+	var filteredKeys [][]byte
 
 	for _, candidateDatastoreKey := range results[0].CandidateDatastoreKeys {
-		if strings.Index(convert.BytesToString(candidateDatastoreKey), keyPrefix) == 0 {
-			filteredKeys = append(filteredKeys, convert.BytesToString(candidateDatastoreKey))
+		if isIncluded {
+			if strings.Index(convert.BytesToString(candidateDatastoreKey), keyPrefix) == 0 {
```

Comment:
`strings.Index(convert.BytesToString(candidateDatastoreKey), keyPrefix)` can be easily replaced by an explaining variable:-)

And then the two nested if can be simplified by one with two conditions